src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -1988,6 +2027,152 @@ int wm_vuldet_report_nvd_vulnerabilities(sqlite3 *db, vu_nvd_report **nvd_report
+    if (res != SQLITE_OK) {
+        return wm_vuldet_sql_error(dbCVE, pStmt);
+    } else {
+        sqlite3_bind_text(pStmt, 1, pkg_reference, -1, NULL);
+        sqlite3_bind_text(pStmt, 2, "*", -1, NULL);
+
+        while (SQLITE_ROW == sqlite3_step(pStmt)) {
+            const char *cve = (const char *) sqlite3_column_text(pStmt, 0);
+            const char *START_INCLUDING = (const char *) sqlite3_column_text(pStmt, 1);
+            const char *START_EXCLUDING = (const char *) sqlite3_column_text(pStmt, 2);
+            const char *END_INCLUDING   = (const char *) sqlite3_column_text(pStmt, 3);
+            const char *END_EXCLUDING   = (const char *) sqlite3_column_text(pStmt, 4);
+
+            if (START_INCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_GREATER_THAN_OR_EQUAL], START_INCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                if (!END_INCLUDING && !END_EXCLUDING) {
+                    vulnerable_global = vulnerable = true;
+
+                } else if (END_INCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_LESS_THAN_OR_EQUAL], END_INCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                    vulnerable_global = vulnerable = true;
+
+                } else if (END_EXCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_LESS_THAN], END_EXCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                    vulnerable_global = vulnerable = true;
+                }
+
+            } else if (START_EXCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_GREATER_THAN], START_EXCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                if (!END_INCLUDING && !END_EXCLUDING) {
+                    vulnerable_global = vulnerable = true;
+
+                } else if (END_INCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_LESS_THAN_OR_EQUAL], END_INCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                    vulnerable_global = vulnerable = true;
+
+                } else if (END_EXCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_LESS_THAN], END_EXCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                    vulnerable_global = vulnerable = true;
+                }
+
+            } else { //START == NULL
+                if (END_INCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_LESS_THAN_OR_EQUAL], END_INCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                    vulnerable_global = vulnerable = true;
+
+                } else if (END_EXCLUDING && (wm_checks_package_vulnerability(pkg_version, vu_package_comp[PKG_LESS_THAN], END_EXCLUDING, VER_TYPE_NVD) == VU_VULNERABLE)) {
+                    vulnerable_global = vulnerable = true;
+                }
+            }
+
+            if (vulnerable == true) {
+                os_calloc(1, sizeof(cve_vuln_pkg), newPkg);
```

Comment:
Don't forget to save the architecture.

Reply:
Also, set the `next` pointer to NULL.

Reply:
The memory is set to 0 when `calloc()` is used, therefore the `next` pointer is by default set to NULL (which is the same as an 8bytes cero value).